download-artifact/action/artifact.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (C) 2022 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import os
+import sys
+import tempfile
+from contextlib import contextmanager
+from enum import Enum
+from pathlib import Path
+from typing import Generator, Iterable, Optional
+from zipfile import ZipFile
+
+import httpx
+from dateutil import parser as dateparser
+from pontos.github.actions.core import ActionIO, Console
+from pontos.github.api import JSON, JSON_OBJECT, GitHubRESTApi
+
+
+def is_event(run: JSON_OBJECT, events: Iterable[str]) -> bool:
+    event = run.get("event")
+    return event in events
+
+
+def created_at(run: JSON_OBJECT):
+    iso_time: str = run.get("created_at")
+    return dateparser.isoparse(iso_time)
+
+
+def json_dump(value: JSON) -> str:
+    return json.dumps(value, indent=2)
+
+
+@contextmanager
+def temp_directory() -> Generator[Path, None, None]:
+    """
+    Context Manager to create a temporary directory
+    """
+    temp_dir = tempfile.TemporaryDirectory()
+    dir_path = Path(temp_dir.name)
+    try:
+        yield dir_path
+    finally:
+        temp_dir.cleanup()
+
+
+class WorkflowRunStatus(Enum):
+    ACTION_REQUIRED = "action_required"
+    CANCELLED = "cancelled"
+    COMPLETED = "completed"
+    FAILURE = "failure"
+    IN_PROGRESS = "in_progress"
+    NEUTRAL = "neutral"
+    QUEUED = "queued"
+    REQUESTED = "requested"
+    SKIPPED = "skipped"
+    STALE = "stale"
+    SUCCESS = "success"
+    TIMED_OUT = "timed_out"
+    WAITING = "waiting"
```

Comment:
Wouldn't it make sense to put this in `pontos.github.api.helper` and accessible over `pontos.github.api` in the `__init__.py` file?

Reply:
Yes we should do that.

Reply:
https://github.com/greenbone/pontos/pull/431